custom_components/steam_wishlist/__init__.py

```diff
@@ -20,7 +20,7 @@ async def async_setup_entry(
     url = entry.data["url"]
     # https://store.steampowered.com/wishlist/profiles/<steam-id>/wishlistdata/
     steam_id = url.split("/")[-3]
-    hass.data[DOMAIN][entry.entry_id] = SensorManager(hass, url)
+    hass.data[DOMAIN][entry.entry_id] = SensorManager(hass, entry, url)
```

Comment:
I would recommend passing just the boolean value rather than the entire config entry instance.

```suggestion
    store_all_wishlist_items = entry.options.get("show_all_wishlist_items", False)
    hass.data[DOMAIN][entry.entry_id] = SensorManager(hass, store_all_wishlist_items, url)
```

Reply:
This required to make several more changes for this to work.  I'm happy to say it's all working now during testing.